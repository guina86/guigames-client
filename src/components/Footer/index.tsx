import Heading from 'components/Heading'
import Logo from 'components/Logo'
import Link from 'next/link'
import * as S from './styles'

export type FooterProps = {
  contentTestId?: string
}

const Footer = ({ contentTestId }: FooterProps) => (
  <S.Wrapper>
    <Logo color="black" />
    <S.Content data-testid={contentTestId}>
      <S.Column>
        <Heading color="black" size="small" lineBottom lineColor="secondary">
          Contact
        </Heading>
        <a href="mailto:sac@guigames.com">sac@guigames.com</a>
      </S.Column>

      <S.Column>
        <Heading color="black" size="small" lineBottom lineColor="secondary">
          Follow us
        </Heading>
        <nav aria-label="social media">
          <a href="https://www.instagram.com/gui-games" target="_blank" rel="noopenner, noreferrer">
            Instagram
          </a>
          <a href="https://www.twitter.com/gui-games" target="_blank" rel="noopenner, noreferrer">
            Twitter
          </a>
          <a href="https://www.youtube.com/gui-games" target="_blank" rel="noopenner, noreferrer">
            Youtube
          </a>
          <a href="https://www.facebook.com/gui-games" target="_blank" rel="noopenner, noreferrer">
            Facebook
          </a>
        </nav>
      </S.Column>

      <S.Column>
        <Heading color="black" size="small" lineBottom lineColor="secondary">
          Links
        </Heading>
        <nav aria-labelledby="footer resources">
          <Link href="/">
            <a>Home</a>
          </Link>
          <Link href="/games">
            <a>Store</a>
          </Link>
          <Link href="/search">
            <a>Buscar</a>
          </Link>
        </nav>
      </S.Column>

      <S.Column aria-labelledby="footer-contact">
        <Heading color="black" size="small" lineBottom lineColor="secondary">
          Location
        </Heading>
        <span>Lorem ipsum dolor sit.</span>
        <span>Lorem, ipsum.</span>
        <span>Lorem, ipsum dolor.</span>
      </S.Column>
    </S.Content>
    <S.Copyright>Gui Games 2022 Â© All rights reserved.</S.Copyright>
  </S.Wrapper>
)

export default Footer
