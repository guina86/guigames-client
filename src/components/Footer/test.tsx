import { RenderResult, screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import Footer from '.'

describe('<Footer />', () => {
  const renderSut = (): RenderResult => renderWithTheme(<Footer contentTestId="content" />)

  it('should render the footer', () => {
    const { container } = renderSut()

    expect(screen.getByLabelText(/gui games/i)).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /contact/i })).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /follow us/i })).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /links/i })).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /location/i })).toBeInTheDocument()
    expect(screen.getByText(/gui games 2022 Â© all rights reserved/i)).toBeInTheDocument()
    expect(screen.getByTestId('content')).toHaveStyle({
      'grid-template-columns': 'repeat(2,1fr)'
    })
    expect(screen.getByTestId('content')).toHaveStyleRule(
      'grid-template-columns',
      'repeat(4,1fr)',
      {
        media: '(min-width: 768px)'
      }
    )

    expect(container).toMatchSnapshot()
  })
})
