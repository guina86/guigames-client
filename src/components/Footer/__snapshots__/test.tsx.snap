// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the footer 1`] = `
.c5 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 1.6rem;
}

.c5::after {
  position: absolute;
  left: 0;
  bottom: -1rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #3CD3C1;
}

.c5::after {
  width: 3rem;
}

.c1 {
  padding-top: 0.8rem;
  color: #030517;
  width: 17rem;
  height: 3.4;
}

.c0 .c4 {
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: repeat(2,1fr);
  gap: 3.2rem;
  margin-top: 2.4rem;
}

.c3 a,
.c3 span {
  display: block;
  color: #5c5c5c;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 0.8rem;
  font-size: 1.6rem;
}

.c3 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6 {
  color: #8F8F8F;
  font-size: 1.2rem;
  margin-top: 4.0rem;
  margin-bottom: 2.4rem;
  text-align: center;
}

@media (min-width:768px) {
  .c2 {
    grid-template-columns: repeat(4,1fr);
  }
}

<div>
  <footer
    class="c0"
  >
    <div
      class="c1"
      color="black"
    >
      <svg
        aria-label="Gui Games"
        role="img"
        viewBox="0 -4 369.979 73.354"
        xmlns="http://www.w3.org/2000/svg"
      >
        <lineargradient
          id="gradient_logo"
          x2="0"
          y2="1"
        >
          <stop
            offset="0%"
            stop-color="#f36d72"
          />
          <stop
            offset="100%"
            stop-color="#eb63b3"
          />
        </lineargradient>
        <path
          d="M89.372 15.488C83.027-8.112 68.82-5.162 62.89-1.589c-5.932 3.567-8.911 5.368-15.38 5.368-6.466 0-9.45-1.8-15.378-5.368C26.199-5.162 11.996-8.111 5.65 15.488-.695 39.085-8.555 85.976 24.548 55.232c0 0 8.226-9.301 22.963-9.301 14.121 0 22.965 9.3 22.965 9.3 33.1 30.743 25.238-16.146 18.896-39.743zm-68.097 8a6.908 6.908 0 0 1 0-13.816 6.906 6.906 0 0 1 6.905 6.908 6.907 6.907 0 0 1-6.905 6.908zM35.148 38.69a8.012 8.012 0 1 1 0-16.025 8.012 8.012 0 0 1 0 16.025zM60.05 27.21a4.354 4.354 0 0 1-4.353-4.36 4.352 4.352 0 0 1 4.353-4.353 4.354 4.354 0 0 1 4.359 4.353 4.354 4.354 0 0 1-4.359 4.36zm8.251 8.824a4.359 4.359 0 0 1 0-8.716 4.356 4.356 0 0 1 4.356 4.356 4.356 4.356 0 0 1-4.356 4.36zm0-17.647a4.358 4.358 0 0 1 0-8.716 4.356 4.356 0 0 1 4.356 4.357 4.356 4.356 0 0 1-4.356 4.359zm9.393 8.823a4.354 4.354 0 0 1-4.353-4.36 4.352 4.352 0 0 1 4.353-4.353 4.354 4.354 0 0 1 4.358 4.353 4.355 4.355 0 0 1-4.358 4.36z"
          fill="url(#gradient_logo)"
          xmlns="http://www.w3.org/2000/svg"
        />
        <path
          class="text"
          d="M115 11.32 120.832 3h21.053v62.652h-13.482v-6.608H115V11.32zm25.066 6.1h-9.738v22.765h-3.638v-26.35h13.403V4.739h-18.218l-5.003 7.116v37.292h13.456v6.635h9.738V17.42zm2.847 33.305V3H169.8v56.044h-21.053zm15.276-11.476h-3.639V4.739h-9.79v35.366l6.393 9.042h16.8V4.739h-9.764v34.51zM197.979 3v21.508h-6.901v13.028h6.902v21.508h-27.153V37.537h6.902V24.509h-6.902V3h27.153zm-18.485 37.29h-6.902v8.856h23.622v-8.855h-6.902V13.594h6.902V4.74h-23.622v8.855h6.902v26.698zM218.04 11.32 223.871 3h21.053v62.652h-13.483v-6.608H218.04V11.32zm25.066 6.1h-9.738v22.765h-3.638v-26.35h13.402V4.739h-18.218l-5.002 7.116v37.292h13.456v6.635h9.737V17.42zM251.784 3h21.053v56.045h-26.885V11.32zm5.778 16.105 3.639-5.27v9.577h-3.639v-4.307zm3.638 30.042h9.764V4.739h-18.137l-5.056 7.25v37.158h9.791v-16.72h3.638v16.72zm52.926 9.898h-40.261L273.89 3h34.376l5.859 8.347v47.697zm-25.013-9.898h9.818V13.835h3.638v35.312h9.765l-.027-37.158-5.056-7.25h-31.514v44.408h9.818V13.835h3.558v35.312zm52.926 9.898h-26.886L315.18 3h26.886v20.492h-8.32v14.927h8.293v20.626zm-1.792-45.451V4.739h-23.22v44.408h23.194v-8.962h-13.375v-8.801h5.029v-9.016h-5.03v-8.774h13.403zm18.176 17.79v8.802h-13.456v8.961h16.826l6.367-9.042V22.37h-13.376v-8.774h13.376V4.74h-18.19l-5.003 7.116v19.529h13.456zm-15.329 27.66V11.32L348.926 3h21.053v47.725l-5.832 8.32h-21.053z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="c2"
      data-testid="content"
    >
      <div
        class="c3"
      >
        <h2
          class="c4 c5"
          color="black"
        >
          Contact
        </h2>
        <a
          href="mailto:sac@guigames.com"
        >
          sac@guigames.com
        </a>
      </div>
      <div
        aria-labelledby="social-media"
        class="c3"
      >
        <h2
          class="c4 c5"
          color="black"
        >
          Follow us
        </h2>
        <nav
          id="social-media"
        >
          <a
            href="https://www.instagram.com/gui-games"
            rel="noopenner, noreferrer"
            target="_blank"
          >
            Instagram
          </a>
          <a
            href="https://www.twitter.com/gui-games"
            rel="noopenner, noreferrer"
            target="_blank"
          >
            Twitter
          </a>
          <a
            href="https://www.youtube.com/gui-games"
            rel="noopenner, noreferrer"
            target="_blank"
          >
            Youtube
          </a>
          <a
            href="https://www.facebook.com/gui-games"
            rel="noopenner, noreferrer"
            target="_blank"
          >
            Facebook
          </a>
        </nav>
      </div>
      <div
        aria-labelledby="resources"
        class="c3"
      >
        <h2
          class="c4 c5"
          color="black"
        >
          Links
        </h2>
        <nav
          id="resources"
        >
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="/games"
          >
            Store
          </a>
          <a
            href="/search"
          >
            Buscar
          </a>
        </nav>
      </div>
      <div
        aria-label="contact"
        class="c3"
      >
        <h2
          class="c4 c5"
          color="black"
        >
          Location
        </h2>
        <span>
          Lorem ipsum dolor sit.
        </span>
        <span>
          Lorem, ipsum.
        </span>
        <span>
          Lorem, ipsum dolor.
        </span>
      </div>
    </div>
    <div
      class="c6"
    >
      Gui Games 2022 Â© All rights reserved.
    </div>
  </footer>
</div>
`;
